client/e2e/poms/operations.ts

```diff
@@ -25,7 +55,84 @@ export class OperationsPOM {
 
   async urlIsCorrect() {
     const path = this.url;
-    const currentUrl = await this.page.url();
-    await expect(currentUrl.toLowerCase()).toMatch(path.toLowerCase());
+    const currentUrl = this.page.url();
+    expect(currentUrl.toLowerCase()).toMatch(path.toLowerCase());
+  }
+
+  async operationsTableIsVisible() {
+    await this.page.waitForSelector(".MuiDataGrid-root");
+  }
+
+  async operationsViewIsCorrect(role: string, expectedStatuses: string[]) {
+    switch (role) {
+      case UserRole.INDUSTRY_USER_ADMIN:
+      case UserRole.INDUSTRY_USER:
+        // later
+        break;
+      default:
+        await expect(this.page.getByText(this.internalNote)).toBeVisible();
+        await expect(this.buttonAdd).not.toBeVisible();
+        break;
+    }
+
+    // Check if the statuses are visible
+    for (const status of expectedStatuses) {
+      await expect(
+        this.page.getByRole("cell", { name: status }).first(),
+      ).toBeVisible();
+    }
+  }
+
+  // 🔩 Below functions are specific to operation detail page
+  async clickExpandAllButton() {
+    // Click Expand All button
+    await this.page.getByRole("button", { name: "Expand All" }).click();
+    const allFormFields = await getAllFormInputs(this.page);
+    for (const field of allFormFields) {
+      // Last input is the file widget input, which is not visible
+      if (field === allFormFields[allFormFields.length - 1]) {
+        await expect(field).not.toBeVisible();
+      } else await expect(field).toBeVisible();
+    }
+  }
+
+  async clickCollapseAllButton() {
+    // Click Collapse All button
+    await this.page.getByRole("button", { name: "Collapse All" }).click();
+    const allFormFields = await getAllFormInputs(this.page);
+    for (const field of allFormFields) await expect(field).not.toBeVisible();
+  }
+
+  async clickOperationsLink() {
+    // Click Operations link on the breadcrumb and wait for the operations table to load
+    await this.page.getByRole("link", { name: "Operations" }).click();
+    await this.page.waitForSelector(".MuiDataGrid-root");
+  }
+
+  async getRequestChangesButton() {
+    const requestChangesButton = this.page.locator(
+      "button[aria-label='Request Changes']",
+    );
+    return requestChangesButton;
+  }
+
+  async getConfirmChangeRequestButton() {
+    return this.page.locator("button[aria-label='Confirm Change Request']");
+  }
+
+  async getCancelChangeRequestButton() {
+    return this.page.locator("button[aria-label='Cancel Change Request']");
+  }
+
+  async getUndoChangeRequestButton() {
+    return this.page.locator("button[aria-label='Undo Request Changes']");
+  }
+
+  async operationApprovedMessageIsVisible() {
+    await this.operationApprovedMessage.waitFor({ state: "visible" });
+  }
```

Comment:
I was having some trouble with confirmation messages in the industry_user test and in the end I got it to work with this:
```
await expect(
      page.getByText(/has been received and will be reviewed./i),
    ).toBeVisible();
```
Using the entire message seemed to confuse playwright, but just looking for a part of it worked

Reply:
Thank you Brianna, I'm going to give it a shot.